src/main/java/org/example/tms/service/JwtService.java

```diff
@@ -0,0 +1,101 @@
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Service;
+
+import java.security.Key;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+@Getter
+@Setter
+@Service
+public class JwtService {
+
+    @Value("${jwt.secret}")
+    private String accessSecretKey;
+
+    @Value("${jwt.refresh.secret}")
+    private String refreshSecretKey;
+
+    public String generateAccessToken(UserDetails userDetails) {
+        return generateToken(new HashMap<>(), userDetails, ConstantUtil.ACCESS_TOKEN_EXPIRATION, accessSecretKey);
+    }
+
+    public String generateRefreshToken(UserDetails userDetails) {
+        return generateToken(new HashMap<>(), userDetails, ConstantUtil.REFRESH_TOKEN_EXPIRATION, refreshSecretKey);
+    }
+
+    public boolean isAccessTokenValid(String token) {
+        return isTokenValid(token, accessSecretKey);
+    }
+
+    public boolean isRefreshTokenValid(String token) {
+        return isTokenValid(token, refreshSecretKey);
+    }
+
+    public String extractUsername(String token, String key) {
+        return extractClaim(token, Claims::getSubject, key);
+    }
+
+    private boolean isTokenValid(String token, String key) {
+        return !isTokenExpired(token, key);
+    }
+
+    private boolean isTokenExpired(String token, String key) {
+        try {
+            return extractExpiration(token, key).before(new Date());
+        } catch (ExpiredJwtException e) {
+            return true;
+        }
+    }
+
+    private Date extractExpiration(String token, String key) {
+        return extractClaim(token, Claims::getExpiration, key);
+    }
+
+    private <T> T extractClaim(String token, Function<Claims, T> claimsResolver, String key) {
+        Claims claims = extractAllClaims(token, key);
+        return claimsResolver.apply(claims);
+    }
+
+    private Claims extractAllClaims(String token, String key) {
+        return Jwts.parserBuilder()
+                .setSigningKey(getSignInKey(key))
+                .build()
+                .parseClaimsJws(token)
+                .getBody();
+    }
+
+    private Key getSignInKey(String key) {
+        byte[] keyBytes = Decoders.BASE64.decode(key);
+        return Keys.hmacShaKeyFor(keyBytes);
+    }
+
+    private String generateToken(Map<String, Object> extraClaims, UserDetails userDetails, int expirationMillis,
+                                 String key) {
+        Date now = new Date();
+        Date expiration = new Date(now.getTime() + expirationMillis);
+
+        return Jwts.builder()
+                .setClaims(extraClaims)
+                .setSubject(userDetails.getUsername())
+                .setIssuedAt(now)
+                .setExpiration(expiration)
+                .signWith(getSignInKey(key), SignatureAlgorithm.HS256)
+                .compact();
+    }
```

Comment:
i think better to create utility class for private methods like this, @Caromeen what do u think?